### PR TITLE
Add GHA for Dead Code Analysis

### PR DESCRIPTION
**Pull request by SimonBrandner**
Towards https://github.com/vector-im/element-web/issues/21969

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: .github/workflows/static_analysis.yaml [hunk 1]
```diff
@@ -74,3 +74,21 @@ jobs:
...
+        with:
+          cache: 'yarn'
+
+      - name: Install Deps
+        run: "scripts/ci/layered.sh"
+
+      - name: Dead Code Analysis
+        run: |
+          cd element-web 
+          yarn run analyse:unused-exports
```
Comment (t3chguy) [inline .github/workflows/static_analysis.yaml:94]: does this script exit with a non-zero code when there is dead code?

Comment (SimonBrandner) [reply .github/workflows/static_analysis.yaml:94]: I do believe it does

**Comment by t3chguy** [comment]
Not sure how valid `Unused exports found!` is given surely it also finds unused unexported code

**Review by dbkr — APPROVED** [review]

**Comment by t3chguy** [comment]
@SimonBrandner this won't auto-merge

1. X-Blocked
2. Unused Exports Found

**Comment by SimonBrandner** [comment]
> @SimonBrandner this won't auto-merge
> 
> 1. X-Blocked
> 2. Unused Exports Found

I am aware :D

**Pull request merged**